homeassistant/components/adguard/__init__.py

```diff
@@ -0,0 +1,197 @@
+"""Support for AdGuard Home."""
+import logging
+from typing import Any, Dict
+
+import voluptuous as vol
+
+from homeassistant.components.adguard.const import (
+    CONF_FORCE, DATA_ADGUARD_CLIENT, DATA_ADGUARD_VERION, DOMAIN,
+    SERVICE_ADD_URL, SERVICE_DISABLE_URL, SERVICE_ENABLE_URL, SERVICE_REFRESH,
+    SERVICE_REMOVE_URL)
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, CONF_PASSWORD, CONF_PORT, CONF_SSL, CONF_URL,
+    CONF_USERNAME, CONF_VERIFY_SSL)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+_LOGGER = logging.getLogger(__name__)
+
+SERVICE_URL_SCHEMA = vol.Schema({vol.Required(CONF_URL): cv.url})
+SERVICE_ADD_URL_SCHEMA = vol.Schema(
+    {vol.Required(CONF_NAME): cv.string, vol.Required(CONF_URL): cv.url}
+)
+SERVICE_REFRESH_SCHEMA = vol.Schema(
+    {vol.Optional(CONF_FORCE, default=False): cv.boolean}
+)
+
+
+async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
+    """Set up the AdGuard Home components."""
+    return True
+
+
+async def async_setup_entry(
+        hass: HomeAssistantType, entry: ConfigType
+) -> bool:
+    """Set up AdGuard Home from a config entry."""
+    from adguardhome import AdGuardHome
+
+    session = async_get_clientsession(hass, entry.data[CONF_VERIFY_SSL])
+    adguard = AdGuardHome(
+        entry.data[CONF_HOST],
+        port=entry.data[CONF_PORT],
+        username=entry.data[CONF_USERNAME],
+        password=entry.data[CONF_PASSWORD],
+        tls=entry.data[CONF_SSL],
+        verify_ssl=entry.data[CONF_VERIFY_SSL],
+        loop=hass.loop,
+        session=session,
+    )
+
+    hass.data.setdefault(DOMAIN, {})[DATA_ADGUARD_CLIENT] = adguard
+
+    for component in 'sensor', 'switch':
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    def add_url(call) -> None:
+        """Service call to add a new filter subscription to AdGuard Home."""
+        adguard = hass.data[DOMAIN][DATA_ADGUARD_CLIENT]
```

Comment:
FYI, you can also use the var from the outer scope directly - so removing this line will do the exact same.

Reply:
👍

Reply:
Addressed on all occurrences. Thanks!